core/crypto/signed_hdr.c

```diff
@@ -60,6 +60,57 @@ static bool is_weak_hash_algo(uint32_t algo)
 	       algo == TEE_ALG_SHA224;
 }
 
+/*
+ * Checks for key sizes in bits, that are either depracted or put into legacy
+ * by national standards bodies. For deprecated key sizes return true. For
+ * legacy key sizes, return false, but print an informative message if
+ * CFG_WARN_INSECURE was set when compiling the TEE.
+ */
+static bool is_weak_key_size(uint32_t algo, size_t key_size_bits)
+{
+	switch (algo) {
```

Comment:
Do you mean `if (TEE_ALG_GET_MAIN_ALG(algo) == TEE_MAIN_ALGO_RSA))`?

Reply:
If that as we works as a "catch-all" for all of the ones below, then absolutely yes! Can you confirm that this is the case?

Reply:
Yes, and this is also a macro that we maintain and extend as needed, so we should normally not need to revisit this function if another RSA algorithm is ever added.